packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1408,7 +1429,7 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
           onSubmitted: widget.onSubmitted,
           onAppPrivateCommand: widget.onAppPrivateCommand,
           onSelectionHandleTapped: _handleSelectionHandleTapped,
-          onTapOutside: widget.onTapOutside,
+          onTapOutside: _handleOnTapOutside(),
```

Comment:
Actually I think `_handleOnTapOutside` logic should actually go in `EditableText` since `onTapOutside` is actually being piped through to `EditableText` and the documentation for `onTapOutside` lives there.